docs/source/llms/openai/autologging.rst

```diff
@@ -0,0 +1,184 @@
+MLflow OpenAI Autologging
+=========================
+
+The OpenAI flavor for MLflow supports autologging to ensure that experimentation, testing, and validation of your ideas can be captured dynamically without 
+having to wrap your code with logging boilerplate. 
+
+In this guide, we'll discuss some of the key features that are available in the autologging feature. 
+
+.. contents:: Table of Contents
+    :local:
+    :depth: 1
+
+Quickstart
+----------
+To get started with MLflow's OpenAI autologging, you simply need to call :py:func:`mlflow.openai.autolog` at the beginning of your script or notebook. 
+Enabling autologging with no argument overrides will behave as the ``default`` configuration in the table in the next section. Overriding any of these settings 
+will allow you to log additional elements. 
+
+.. tip::
+    The only element that is **enabled by default** when autologging is activated is the recording of trace information. You can read more about MLflow tracing 
+    `here <../tracing/index.html>`_. 
+
+.. code-block:: python
+
+    import os
+    import openai
+    import mlflow
+
+    mlflow.openai.autolog()
+
+    openai_client = openai.OpenAI(api_key=os.environ.get("OPENAI_API_KEY"))
+
+    # Use the openai_client and traces will be logged to the active experiment
+    ...
+
+Configuration of OpenAI Autologging
+-----------------------------------
+
+MLflow OpenAI autologging can log various information about the model and its inference. **By default, only trace logging is enabled**, but you can enable 
+autologging of other information by setting the corresponding parameters when calling :py:func:`mlflow.openai.autolog()`. 
+
+The available options and their default values are shown below. To learn more about additional parameters, see the API documentation.
+
+.. list-table::
+    :widths: 20 20 30 30
+    :header-rows: 1
+
+    * - Target
+      - Default
+      - Parameter
+      - Description
+    * - Traces
+      - ``true``
+      - ``log_traces``
+      - Whether to generate and log traces for the model. See `MLflow Tracing <../tracing/index.html>`_ for more details about the tracing feature.
+    * - Model Artifacts
+      - ``false``
+      - ``log_models``
+      - If set to ``True``, the OpenAI model will be logged when it is invoked. 
+    * - Model Signatures
+      - ``false``
+      - ``log_model_signatures``
+      - If set to ``True``, :py:class:`ModelSignatures <mlflow.models.ModelSignature>` describing model inputs and outputs are collected and logged along with OpenAI model artifacts during inference. This option is only available when ``log_models`` is enabled.
+    * - Input Example
+      - ``false``
+      - ``log_input_examples``
+      - If set to ``True``, input examples from inference data are collected and logged along with OpenAI model artifacts during inference. This option is only available when ``log_models`` is enabled.
+    * - Datasets
+      - ``false``
+      - ``log_datasets``
```

Comment:
`log_datasets` is actually no-op for OpenAI (similar to langchain). Can we remove this row?